cmd/gimlet/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! ## `humility gimlet`
+//!
+//! `humility gimlet` contacts a (recent firmware version) Gimlet over the
+//! management network and extracts certain live diagnostic information.
+//!
+//! ```console
+//! $ humility gimlet --ip fe80::0c1d:9aff:fe64:b8c2%en0 read-seq-regs
+//! ```
+//!
+//! For a complete list of subcommands, use `humility gimlet --help`.
+
+use std::net::{SocketAddrV6, UdpSocket};
+use std::time::Duration;
+
+use anyhow::{bail, Context, Result};
+use clap::{ArgGroup, IntoApp, Parser};
+use humility::net::decode_iface;
+use humility_cli::{ExecutionContext, Subcommand};
+use humility_cmd::{Archive, Command, CommandKind, Dumper};
+
+#[derive(Parser, Debug)]
+#[clap(
+    name = "gimlet", about = env!("CARGO_PKG_DESCRIPTION"),
+    group = ArgGroup::new("target").multiple(false)
+)]
+struct Args {
+    /// sets timeout
+    #[clap(
+        long, short = 'T', default_value_t = 2000, value_name = "timeout_ms",
+        parse(try_from_str = parse_int::parse)
+    )]
+    timeout: u32,
+
+    /// IPv6 address, e.g. `fe80::0c1d:9aff:fe64:b8c2%en0`
+    #[clap(
+        long,
+        env = "HUMILITY_RPC_IP",
+        group = "target",
+        use_value_delimiter = true
+    )]
+    ip: String,
+
+    #[clap(subcommand)]
+    cmd: SubCmd,
+}
+
+#[derive(Parser, Debug)]
+enum SubCmd {
+    /// Read the current contents of the sequencer FPGA registers.
+    ReadSeqRegs,
+}
+
+pub struct Client {
+    socket: UdpSocket,
+    // Allocating this overly-large buffer because we're on the big computer
+    // where RAM is cheap, and it's less likely to need adjustment this way.
+    buf: [u8; 1024],
+}
+
+impl Client {
+    pub fn new(ip: &str, timeout: Duration) -> Result<Self> {
+        let Some((ip, scope)) = ip.rsplit_once('%') else {
+            bail!("Missing scope id in IP (e.g. '%en0')")
+        };
+
+        let scopeid = decode_iface(scope)?;
+
+        let dest = SocketAddrV6::new(
+            ip.parse()?,
+            23547,
```

Comment:
Add a comment here that this is a hard-coded socket address for the `gimlet-inspector` task

Reply:
I can do 'ya one better:
- Extracted a constant
- Added a comment explaining that it corresponds to the number in the TOML
- Also made it overrideable using a command line argument.